airflow/providers/microsoft/azure/sensors/azure_data_factory.py

```diff
@@ -0,0 +1,81 @@
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from typing import Dict, Optional
+
+from airflow.exceptions import AirflowException
+from airflow.providers.microsoft.azure.hooks.azure_data_factory import (
+    AzureDataFactoryHook,
+    AzureDataFactoryPipelineRunStatus,
+)
+from airflow.sensors.base import BaseSensorOperator
+
+
+class AzureDataFactoryPipelineRunStatusSensor(BaseSensorOperator):
+    """
+    Checks the status of a pipeline run.
+
+    :param conn_id: The connection identifier for connecting to Azure Data Factory.
+    :type conn_id: str
+    :param run_id: The pipeline run identifier.
+    :type run_id: str
+    :param resource_group_name: The resource group name.
+    :type resource_group_name: str
+    :param factory_name: The data factory name.
+    :type factory_name: str
+    """
+
+    template_fields = ("conn_id", "resource_group_name", "factory_name", "run_id")
+
+    def __init__(
+        self,
+        *,
+        conn_id: str,
+        run_id: str,
+        resource_group_name: Optional[str] = None,
+        factory_name: Optional[str] = None,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.conn_id = conn_id
+        self.run_id = run_id
+        self.resource_group_name = resource_group_name
+        self.factory_name = factory_name
+
+    def poke(self, context: Dict) -> bool:
+        self.log.info(f"Checking the status for pipeline run {self.run_id}.")
+        self.hook = AzureDataFactoryHook(conn_id=self.conn_id)
+        pipeline_run = self.hook.get_pipeline_run(
+            run_id=self.run_id,
+            factory_name=self.factory_name,
+            resource_group_name=self.resource_group_name,
+        )
+        pipeline_run_status = pipeline_run.status
+        self.log.info(f"Current status for pipeline run {self.run_id}: {pipeline_run_status}.")
+
+        if pipeline_run_status == AzureDataFactoryPipelineRunStatus.SUCCEEDED:
+            self.log.info(f"The pipeline run {self.run_id} has succeeded.")
+            return True
+        elif pipeline_run_status in {
+            AzureDataFactoryPipelineRunStatus.FAILED,
+            AzureDataFactoryPipelineRunStatus.CANCELLED,
+        }:
+            raise AirflowException(
+                f"Pipeline run {self.run_id} is in a negative terminal status: {pipeline_run_status}"
+            )
+
+        return False
```

Comment:
Again would recommend to use a function that returns a job status by id/name (same comment that I gave for the operat) take a look at TableauJobStatusSensor. The code is much more clean & simple when the logic is in the hook.

https://github.com/apache/airflow/blob/95279bdb8989fc0e433799792dab2ed38373e1f8/airflow/providers/tableau/sensors/tableau_job_status.py#L66-L73